src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,286 @@
+            message = `Color value ${color} should be of format 'rgb(115,159,159)'`;
+        } else if (color && this.props.format === "rgba" && (color.toLowerCase().indexOf("rgba") === -1)) {
+            message = `Color value ${color} should be of format 'rgba(195,226,226,1)'`;
+        }
+
+        return {
+            alertMessage: message,
+            color,
+            displayColorPicker: this.state.displayColorPicker
+        };
+    }
+
+    private handleInputChange = (event: any) => {
+        const newColor = event.target.value as string;
+        if (newColor) {
+            this.colorChanged = this.getValue(this.props.mxObject) !== newColor;
+            const state = this.validateColor(event.target.value);
+            if (!state.alertMessage && this.props.mxObject) {
+                this.props.mxObject.set(this.props.colorAttribute, newColor);
+                this.handleOnChange();
+            }
+            this.setState(state);
+        } else {
+            this.setState({ alertMessage: "Invalid color", color: event.target.value });
+        }
+    }
+
+    private handleOnChange = () => {
+        const { mxform, mxObject, onChangeEvent, onChangeMicroflow, onChangeNanoflow, onChangePage } = this.props;
+        if (mxObject && ((this.previousColor !== this.state.color) || this.colorChanged)) {
+            const context = new mendix.lib.MxContext();
+            context.setContext(mxObject.getEntity(), mxObject.getGuid());
+            if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+                window.mx.ui.action(onChangeMicroflow, {
+                    error: error => window.mx.ui.error(`Error while executing microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                    params: {
+                        applyto: "selection",
+                        guids: [ mxObject.getGuid() ]
+                    },
+                    origin: mxform
+                });
+            } else if (onChangeEvent === "showPage" && onChangePage) {
+                window.mx.ui.openForm(onChangePage, {
+                    context,
+                    error: error => window.mx.ui.error(`Error while opening page ${onChangePage}: ${error.message}`),
+                    location: this.props.openPageLocation
+                });
+            } else if (onChangeEvent === "callNanoflow" && onChangeNanoflow.nanoflow) {
+                window.mx.data.callNanoflow({
+                    context,
+                    error: error => window.mx.ui.error(`Error while executing the onchange nanoflow: ${error.message}`),
+                    nanoflow: onChangeNanoflow,
+                    origin: mxform
+                });
+            }
+        }
+    }
+
+    public static validateProps(props: ColorPickerContainerProps): string {
+        const message: string[] = [];
+        if (props.onChangeEvent === "callMicroflow" && !props.onChangeMicroflow) {
```

Comment:
Change order of if statements, so it matches the configuration, page, microflow, nanoflow

Reply:
Done